Add helm test for Tumbleweed

ilausuch opened the pull request:
This new test checks the main operations of helm, and deploys
an apache on EKS

- Related ticket: https://progress.opensuse.org/issues/108572
- Verification run: http://copland.arch.suse.de/tests/1276 (Tumbleweed)
- http://copland.arch.suse.de/tests/1275 (Leap 15.3)

pdostal reviewed (approved):
This is so easy to read, I like it, it is cool! :+1:

ilausuch commented on tests/containers/helm.pm:
```diff
@@ -0,0 +1,87 @@
+use registration qw(add_suseconnect_product get_addon_fullname);
+
+sub install_kubectl {
+    my $version = get_var('PUBLICCLOUD_KUBECTL_VERSION', 'v1.22.2');
+    assert_script_run("curl -LO https://dl.k8s.io/release/$version/bin/linux/amd64/kubectl");
```
mmm, yesss maybe is only on sle we don't have it, in other tests we are doing this... I have to do the version for sle too, so I will save this for sle and if we have kubectl on tumbleweed let's use it. Thanks

ilausuch replied:
And thanks Pavel :)

ilausuch replied:
done

ilausuch commented:
I fixed the installation of kubectl 
And I changed the assert_script_run for checking the connectivity with apache. Now I am using script_retry that allows to do several retries until the service is ready

ilausuch commented:
I checked that the same code works with Leap too

jlausuch commented on tests/containers/helm.pm:
```diff
@@ -0,0 +1,75 @@
+    zypper_call("in -y jq aws-cli helm podman kubernetes-client");
+
+    # Check helm version
+    assert_script_run("helm version");
```
I guess this line is redundant as it's executed below.

ilausuch replied:
done

jlausuch commented on tests/containers/helm.pm:
```diff
@@ -0,0 +1,75 @@
+    assert_script_run("helm version");
+    record_info('Helm version', script_output("helm version"));
+
+    # Connexion to AWS
```
```suggestion
    # Connection to AWS
```

ilausuch replied:
done

jlausuch commented on tests/containers/helm.pm:
```diff
@@ -0,0 +1,75 @@
+    assert_script_run("cat \$HOME/.kube/config");
+    assert_script_run("kubectl get nodes");
+
+    # Ensure is deleted
```
just curious, why deleting something that hasn't been created before?

ilausuch replied:
If the previous step of deletion has failed then this test never will start and manually we'll have to clean up the k8s. 
But there is an other solution, and that is to set up a new name every time, this allows to have some tests in parallel. Changing that

jlausuch replied:
This is dangerous, what if 2 tests run at the same time?

ilausuch replied:
now, no problem

jlausuch replied:
if we are generating an unique deployment name, do we still need this?

jlausuch commented on tests/containers/helm.pm:
```diff
@@ -0,0 +1,75 @@
+
+    # Test
+    enter_cmd("kubectl port-forward  services/$self->{deployment_name} 10001:80 &");
+    script_retry("curl http://localhost:10001", delay => 5, retry => 3);
```
Good idea. Since we have recently expected some network slowdown, maybe we could increase the retries and the delay. Something like `delay => 30, retry => 5`? Just to be in the safe side.

ilausuch replied:
done

jlausuch commented on tests/containers/helm.pm:
```diff
@@ -0,0 +1,75 @@
+
+sub post_run_hook {
+    my ($self) = @_;
+    assert_script_run("helm delete $self->{deployment_name}");
```
I would move this to the test flow instead of the post hook, as it's kind of part of the package test.

ilausuch replied:
Ok, but only for the post hook, we need to keep it for the post_fail_hook

jlausuch replied:
yes, the post fail is fine as it is

ilausuch replied:
done

jlausuch commented on tests/containers/helm.pm:
```diff
@@ -0,0 +1,75 @@
+
+sub post_fail_hook {
+    my ($self) = @_;
+    assert_script_run("helm delete $self->{deployment_name}");
```
```suggestion
    script_run("helm delete $self->{deployment_name}");
```
Just to not fail the test here, in case the test fails before creating the heml chart.

ilausuch replied:
mmm I show that in a post_fail_hook, this line is executed inside a post_fail_hook https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/f02af1d6b443591159e95e0ca0ab330ec2f7b308/lib/publiccloud/eks.pm#L33 should we remove the assert_script_run from there too?

jlausuch replied:
Yes, probably `script_run` fits better in these cases... I'm not 100% sure how post_fail hooks treat failures.... 
Anyway `eks.pm` can be changed in a different PR.

ilausuch replied:
done

jlausuch commented on schedule/containers/helm.yaml:
```diff
@@ -0,0 +1,7 @@
+name: helm
+description: |
+  helm test
+schedule:
```
after a chat, agreed to use `main_containers.pm` instead of yaml schedule.

ilausuch replied:
done

ilausuch commented:
I change the name of the deployment and now is dynamic
```$self->{deployment_name} = "apache-" . get_current_job_id();```

mloviska commented on tests/containers/helm.pm:
```diff
@@ -0,0 +1,73 @@
+
+    # Add repo, search and show values
+    assert_script_run("helm repo add bitnami https://charts.bitnami.com/bitnami");
+    assert_script_run("helm search repo apache");
```
Just out of curiosity, should not the test try `helm repo update` before search and install ?

ilausuch replied:
yes, we can add this. Thanks

ilausuch replied:
done and testing
